pkg/detectors/gcp/gcp.go

```diff
@@ -89,30 +82,30 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		}
 
 		credBytes, _ := json.Marshal(creds)
-
 		s := detectors.Result{
 			DetectorType: detectorspb.DetectorType_GCP,
 			Raw:          raw,
 			RawV2:        credBytes,
 			Redacted:     creds.ClientEmail,
-		}
-		// Set the RotationGuideURL in the ExtraData
-		s.ExtraData = map[string]string{
-			"rotation_guide": "https://howtorotate.com/docs/tutorials/gcp/",
-			"project":        creds.ProjectID,
+			ExtraData: map[string]string{
+				"rotation_guide": "https://howtorotate.com/docs/tutorials/gcp/",
+				"project":        creds.ProjectID,
+			},
 		}
 
 		if verify {
 			credentials, err := google.CredentialsFromJSON(ctx, credBytes, "https://www.googleapis.com/auth/cloud-platform")
 			if err != nil {
+				s.SetVerificationError(err)
 				continue
 			}
-			if credentials != nil {
```

Comment:
I looked through the code to `google.CredentialsFromJSON` and it doesn't seem possible for `credentials` to be nil if `err` is nil.